File: spec/python/spec/test_eof_exception_bits_be2.py
import unittest
from testformats.eof_exception_bits_be2 import EofExceptionBitsBe2
import kaitaistruct

class TestEofExceptionBitsBe2(unittest.TestCase):
    def test_eof_exception_bits_be2(self):
        with EofExceptionBitsBe2.from_file('src/nav_parent_switch.bin') as r:
            with self.assertRaises(EOFError) as cm:
                r._read()
            self.assertEqual(str(cm.exception), u"requested 3 bytes, but only 2 bytes available")

            self.assertEqual(r._io.pos(), 1)

            self.assertEqual(r.pre_bits, 0x01)
            self.assertIn('start', r._debug['fail_bits'])
            self.assertEqual(r._debug['fail_bits']['start'], 1)
            self.assertFalse(hasattr(r, 'fail_bits'))
            self.assertNotIn('end', r._debug['fail_bits'])

File: spec/java/src/io/kaitai/struct/spec/TestEofExceptionBitsBe2.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.EofExceptionBitsBe2;
import org.testng.annotations.Test;
import static org.testng.Assert.*;

import org.testng.Assert.ThrowingRunnable;
public class TestEofExceptionBitsBe2 extends CommonSpec {
    @Test
    public void testEofExceptionBitsBe2() throws Exception {
        EofExceptionBitsBe2 r = EofExceptionBitsBe2.fromFile(SRC_DIR + "nav_parent_switch.bin");
        assertThrowsEofError(new ThrowingRunnable() {
            @Override
            public void run() throws Throwable {
                r._read();
            }
        });
        assertIntEquals(r._io().pos(), 1);
        assertIntEquals(r.preBits(), 1);
        assertTrue(r._attrStart.containsKey("failBits"));
        assertIntEquals(r._attrStart.get("failBits"), 1);
        assertIntEquals(r.failBits(), 0); // actually uninitialized, hence 0 in Java
        assertFalse(r._attrEnd.containsKey("failBits"));
    }
}
